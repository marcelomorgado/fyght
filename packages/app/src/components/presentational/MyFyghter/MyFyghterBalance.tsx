import React from "react";
import { Spin } from "antd";
import { formatDai } from "../../../helpers";

interface Props {
  balance: BalanceState;
}

export const MyFyghterBalance: React.FC<Props> = ({ balance: { amount, loading } }: Props) => {
  return (
    <>
      <span style={{ color: "blue" }}>{`${formatDai(amount)}`}</span>{" "}
      {loading ? (
        <>
          {" "}
          <Spin size="small" />
        </>
      ) : null}
    </>
  );
};
