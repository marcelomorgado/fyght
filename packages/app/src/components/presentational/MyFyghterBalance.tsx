import React from "react";
import { formatDai } from "../../helpers";
import { BigNumber } from "ethers";

interface Props {
  value: BigNumber;
}

export const MyFyghterBalance: React.FC<Props> = ({ value }: Props) => {
  return <span style={{ color: "blue" }}>{`${formatDai(value)}`}</span>;
};
