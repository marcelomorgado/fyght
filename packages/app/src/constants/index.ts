export enum Skin {
  NAKED = "naked",
  NORMAL_GUY = "normal_guy",
  KARATE_KID = "karate_kid",
  JAPONESE = "japonese",
  MONK = "monk",
  NINJA = "ninja",
  NO_ONE = "no_one",
  DAEMON = "daemon",
  MASTER = "master",
}

export enum AvatarSize {
  SMALL = "small",
  MEDIUM = "medium",
}
