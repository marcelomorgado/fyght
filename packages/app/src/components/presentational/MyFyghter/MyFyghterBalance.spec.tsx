import * as React from "react";
import { shallow } from "enzyme";
import { MyFyghterBalance } from "./MyFyghterBalance";
import { ONE } from "../../../constants";

describe("MyFyghterBalance", () => {
  test("should render the component", () => {
    expect(shallow(<MyFyghterBalance value={ONE} />)).toMatchSnapshot();
  });
});
