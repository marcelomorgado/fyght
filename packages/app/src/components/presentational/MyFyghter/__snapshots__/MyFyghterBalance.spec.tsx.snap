// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFyghterBalance should render the component 1`] = `
<span
  style={
    Object {
      "color": "blue",
    }
  }
>
  $1.00
</span>
`;
